### DISCUSSION TITLE
Volt rules with closure : array_merge(): Argument #2 must be of type array, Closure given

### DISCUSSION THREAD

**Discussion by LeDave** [livewire/livewire#7117] [1. Help]
Hello,

I am trying to use a closure to validate my rules in functionnal volt, but I receive this error : array_merge(): Argument #2 must be of type array, Closure given

I tried to use the example from the docs:

```
rules(fn () => [
    'name' => ['required', 'min:6'],
    'email' => ['required', 'email', 'not_in:'.Auth::user()->email]
]);
```

When I look at the rules function it seems like closures are not supported:

```
function rules(mixed ...$rules): RuleOptions
{
    if (count($rules) === 1 && array_key_exists(0, $rules)) {
        $rules = $rules[0];
    }

    CompileContext::instance()->rules = array_merge(
        CompileContext::instance()->rules,
        $rules,
    );

    return new RuleOptions;
}
```

Can somebody help, please?

**Comment by danie-ramdhani** [comment]
you can't use `array_merge` if one of the parameters is not an array.

in example:

```php
$foo = fn () => ['foo', 'bar'];
dd($foo);
// the result is closure, not array:
// Closure() {
//    class: "Path\To\YourClass"
//    this: Path\To\YourClass
// }
```

but, you can get the return value from the  closure like this:

```php
$foo = fn () => ['foo', 'bar'];
dd($foo());
// the result is array:
// array:2 [▼
//   0 => "foo"
//   1 => "bar"
// ]
```

**Reply by LeDave** [reply]
Yes thank you. But the second code is coming from the Volt package. At first glance it looks like rules cannot handle closures, so the example from the docs (https://livewire.laravel.com/docs/volt#validation) can't work.
